src/type-layout.md

```diff
@@ -226,6 +228,15 @@ for field in struct.fields_in_declaration_order() {
 struct.size = current_offset + padding_needed_for(current_offset, struct.alignment);
 ```
 
+<div class="warning">
+
+Warning: The function used to calculate the needed padding in this pseudocode is
+a naive implementation that doesn't account for overflow issues. For a rigorous
```

Comment:
The pseudocode as a whole does not account for overflows -- it's not just the padding part.

IMO this should recommend to construct the layout using `extend` to add each field, and using `pad_to_align` for the final padding.

Reply:
I wonder if it might just be better to just make the link part a general recommendation to use `std::alloc::Layout` to work with layouts in real code.

Reply:
I mean, yes, that was basically the point of this warning in my eyes.